security/hcaptcha.go

```diff
@@ -0,0 +1,87 @@
+package security
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type CaptchaResponse struct {
+	Token string `json:"hcaptcha_token"`
+}
+
+type VerificationResponse struct {
+	Success    bool     `json:"success"`
+	ErrorCodes []string `json:"error-codes"`
+	Hostname   string   `json:"hostname"`
+}
+
+type VerificationResult int
+const (
+	UserRequestFailed VerificationResult = iota
+	VerificationProcessFailure
+	SuccessfullyVerified
+)
+var Client *http.Client
+
+func init() {
+	// TODO (darora): make timeout configurable
+	Client = &http.Client{Timeout: 10 * time.Second}
+}
+
+func VerifyRequest(r *http.Request, secretKey string) (VerificationResult, error) {
+	res := CaptchaResponse{}
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return UserRequestFailed, err
+	}
+	r.Body.Close()
```

Comment:
nit: defer this?

Reply:
There's no reason to defer it, it's replaced in the next statement, and has already been consumed